static/js/containers/CreatePostPage.js

```diff
@@ -0,0 +1,137 @@
+// @flow
+/* global history: false */
+import React from 'react';
+import { connect } from 'react-redux';
+import R from 'ramda';
+import { FETCH_SUCCESS } from 'redux-hammock/constants';
+
+import Card from '../components/Card';
+import ChannelSidebar from '../components/ChannelSidebar';
+import PostList from '../components/PostList';
+import ChannelBreadcrumbs from '../components/ChannelBreadcrumbs';
+import { actions } from '../actions';
+
+import type { Channel } from '../flow/discussionTypes';
+import type { Dispatch } from 'redux';
+import type { Match } from 'react-router';
+import {
+  reset,
+  setIsText,
+  setTitle,
+  setText,
+  setUrl,
+} from "../actions/createNewPost";
+
+
+type CreatePostPageProps = {
+  match:            Match,
+  dispatch:         Dispatch,
+  channel:          Channel,
+};
+
+const getChannelName = (props: CreatePostPageProps): string => (
+  props.match.params.channelName
+);
+
+class CreatePostPage extends React.Component {
+  props: CreatePostPageProps;
+
+  componentWillMount() {
+    const { dispatch } = this.props;
+    const channelName = getChannelName(this.props);
+    if (!actions.channels.loaded && !actions.channels.processing) {
```

Comment:
I think you meant to pull `channels` from `this.props` here, not check `actions.channels.loaded`.